core/utils/silly-i18n/cypress/integration/browser_spec.js

```diff
@@ -1,17 +1,41 @@
-import { determineLanguage, I18n } from "../../src/index.js";
+import { determineLanguage, determineLocale } from "../../src/locale.js";
+import { I18n } from "../../src/index.js";
 
 const LANGUAGE = "foo";
+let i18n;
+
+beforeEach(() => {
+    i18n = new I18n(LANGUAGE, {
+        [LANGUAGE]: { quux: { bar: "baz" }, options: { opt: "{{opt}}" } },
+    });
+});
 
 describe("Test language determination", () => {
     it("finds a reasonable two-letter language", () => {
+        expect(determineLocale()).to.match(/^\w{2,}-\w+/);
         expect(determineLanguage()).to.match(/^\w{2,}$/);
     });
 
     it("is able to create the object in a web environment", () => {
-        const i18n = new I18n(LANGUAGE, {
-            [LANGUAGE]: { quux: { bar: "baz" }, options: { opt: "{{opt}}" } },
-        });
-        expect(i18n).to.have.keys(["language", "_translations"]);
+        expect(i18n).to.have.keys(["_locale", "language", "_translations"]);
         expect(i18n.language).to.equal(LANGUAGE);
     });
 });
+
+describe("Test locale numeric options correctly", () => {
+    it("Converts big numbers to locale format", () => {
+        const bigNumber = "1000000.33";
+        const localePairs = {
+            "de-DE": "1.000.000,33",
+            "es-ES": "1.000.000,33",
+            "en-GB": "1,000,000.33",
+        };
+        for (const [locale, l8nString] of Object.entries(localePairs)) {
+            i18n._locale = locale;
+            expect(i18n.locale).to.equal(locale);
+            expect(i18n.t("options:opt", { opt: bigNumber })).to.equal(
```

Comment:
The `options` namespace confused me a bit here - it's not a thing silly-i18n supports natively from how I read the code, but you create an `options` namespace here for the tests. Works for me, though I'd find it clearer if you were to just write `i18n.t("{{opt}}", { opt: bigNumber })).toEqual(`.

Reply:
Keys do have that format, though, because they are all divided in namespaces. I'll try and give them a more contextual name, though.

Reply:
And what you pass to `t` has the format `t(key, options ={}). No possibility of passing the string directly. This is not a templating engine where you can use arbitrary strings, you need to use the strings that have passed on with the constructors.
What you say could make sense with a separate localization-only module, because sometimes you just need to format numbers (say, in a chart or something). That's something I intended to do in the future, creating a separate l12n-only class within this same module; possibly after this is merged (and now I see there's real interest in it)

Reply:
Ah sorry, my bad! Then it's all fine.